src/github.com/couchbase/sync_gateway/db/changes.go

```diff
@@ -403,21 +422,31 @@ func (db *Database) MultiChangesFeed(chans base.Set, options ChangesOptions) (<-
 			default:
 			}
 
+			// Reset added channels
+			addedChannels = nil
+
 			// Before checking again, update the User object in case its channel access has
 			// changed while waiting:
+			base.LogTo("Changes+", "MultiChangesFeed checking for reload..................")
 			if newCount := changeWaiter.CurrentUserCount(); newCount > userChangeCount {
+				var previousChannels channels.TimedSet
 				base.LogTo("Changes+", "MultiChangesFeed reloading user %+v", db.user)
 				userChangeCount = newCount
-				isAdmin := db.user == nil
-				if err := db.ReloadUser(); err != nil {
-					base.Warn("Error reloading user %q: %v", db.user.Name(), err)
-					if !isAdmin {
+
+				if db.user != nil {
+					previousChannels = db.user.InheritedChannels()
+					if err := db.ReloadUser(); err != nil {
+						base.Warn("Error reloading user %q: %v", db.user.Name(), err)
 						change := makeErrorEntry("User not found during reload - terminating changes feed")
 						base.LogTo("Changes+", "User not found during reload - terminating changes feed with entry %+v", change)
 						output <- &change
+						return
 					}
-					return
+					// check whether channels have changed
+					addedChannels = db.user.GetAddedChannels(previousChannels)
+					base.LogTo("Changes+", "New channels found after user reload: %v", addedChannels)
```

Comment:
What do you think about wrapping that log statement in an `if len(addedChannels) > 0` check to reduce log noise?